tests/foreman/ui/test_http_proxy.py

```diff
@@ -67,3 +82,117 @@ def test_positive_create_update_delete(session, module_org, module_loc):
+        organization=[module_org.id],
+        location=[module_loc.id],
+    ).create()
+    http_proxy_url_b = '{}:{}'.format(
+        gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999)
+    )
+    http_proxy_b = entities.HTTPProxy(
+        name=gen_string('alpha', 15),
+        url=http_proxy_url_b,
+        organization=[module_org.id],
+        location=[module_loc.id],
+    ).create()
+    # Create products
+    product_a = entities.Product(organization=module_org.id,).create()
+    product_b = entities.Product(organization=module_org.id,).create()
+    # Create repositories from UI.
+    with session:
+        session.organization.select(org_name=module_org.name)
+        session.location.select(loc_name=module_loc.name)
+        repo_a1_name = gen_string('alpha')
+        session.repository.create(
+            product_a.name,
+            {
+                'name': repo_a1_name,
+                'repo_type': REPO_TYPE['yum'],
+                'repo_content.upstream_url': FAKE_1_YUM_REPO,
+                'repo_content.http_proxy_policy': 'No HTTP Proxy',
+            },
+        )
+        repo_a1_values = session.repository.read(product_a.name, repo_a1_name)
+        assert repo_a1_values['repo_content']['http_proxy_policy'] == 'No HTTP Proxy'
+        repo_a2_name = gen_string('alpha')
+        session.repository.create(
+            product_a.name,
+            {
+                'name': repo_a2_name,
+                'repo_type': REPO_TYPE['yum'],
+                'repo_content.upstream_url': FAKE_1_YUM_REPO,
+                'repo_content.http_proxy_policy': 'Use specific HTTP Proxy',
+                'repo_content.proxy_policy.http_proxy': http_proxy_a.name,
+            },
+        )
+        repo_a2_values = session.repository.read(product_a.name, repo_a2_name)
+        assert repo_a2_values['repo_content'][
+            'http_proxy_policy'
+        ] == 'Use specific HTTP Proxy ({})'.format(http_proxy_a.name)
+        repo_b1_name = gen_string('alpha')
+        session.repository.create(
+            product_b.name,
+            {
+                'name': repo_b1_name,
+                'repo_type': REPO_TYPE['puppet'],
+                'repo_content.upstream_url': FAKE_0_PUPPET_REPO,
+                'repo_content.http_proxy_policy': 'Global Default',
+            },
+        )
+        repo_b1_values = session.repository.read(product_b.name, repo_b1_name)
+        assert 'Global Default' in repo_b1_values['repo_content']['http_proxy_policy']
+        repo_b2_name = gen_string('alpha')
+        session.repository.create(
+            product_b.name,
+            {
+                'name': repo_b2_name,
+                'repo_type': REPO_TYPE['puppet'],
+                'repo_content.upstream_url': FAKE_0_PUPPET_REPO,
+                'repo_content.http_proxy_policy': 'No HTTP Proxy',
+            },
+        )
+        # Add http_proxy to products
+        session.product.search('')
```

Comment:
You shouldn't need to do that - it's `session.product.manage_http_proxy` to figure out that elements to select are not displayed, and reset any filter there might be to make them visible.

I don't want to block your PR on that, so can you open Airgun issue?

Reply:
Created issue https://github.com/SatelliteQE/airgun/issues/465